### ISSUE TITLE
iccv19 light weight face recognition challenge tips

### ISSUE THREAD

**Issue by nttstar** [deepinsight/insightface#632]
**Challenge Information:**

[HomePage](https://ibug.doc.ic.ac.uk/resources/lightweight-face-recognition-challenge-workshop/), please read carefully about the training and testing rules.

**How To Start:**

Please check [https://github.com/deepinsight/insightface/tree/master/iccv19-challenge](https://github.com/deepinsight/insightface/tree/master/iccv19-challenge) for detail.

Any question can be left here.

**Comment by borisgribkov** [comment]
HI @nttstar, regarding the dataset, is it already 5-keypoints aligned? Sorry, have not downloaded it.

**Comment by nttstar** [comment]
@borisgribkov Yes already aligned, so no need to further process on it.

**Comment by zhouwei5113** [comment]
@nttstar what's the difference between ms1m-retinaface-t1.zip and ms1m-v2(emore)?

**Comment by nttstar** [comment]
@zhouwei5113 ms1m-retinaface has different clean list than ms1m-arcface and it has been re-aligned with our RetinaFace detector. You must use ms1m-retinaface to participate in this challenge, all other training-sets including ms1m-arcface are not allowed.

**Comment by leolordwz** [comment]
Hi, @nttstar, could you please share with us how the provided dataset generated from ms-1m aligned dataset. I tried to upscale images in ms-1m aligned dataset to 400x400, but the landmarks do not match. It would be great if you could share with us the preparation process.

**Comment by nttstar** [comment]
@leolordwz it's not allowed for this challenge.

**Comment by chenshen03** [comment]
How can I take part in the iccv19 light weight face recognition challenge?

**Comment by nttstar** [comment]
@chenshen03 Test datasets and test server will be ready soon, any question on training?

**Comment by jamessmith90** [comment]
@nttstar Can you provide more info about RetinaFace Detector ?

**Comment by LCorleone** [comment]
@nttstar 
Is the test set aligned by RetinaFace？ If not, will you release the RetinaFace Detector？
Many thanks！

**Comment by IrvingShu** [comment]
ResNet100FC-IR contains the fc7 layer ? if remove last fc7 layer, the flops is  12 GFLOPs.

**Comment by BingranHu** [comment]
Hi@nttstar, 
I have trained a R100SE model  with arcface loss on the ms1m-retinaface dataset very recently. 
And the highest accuracy on the lfw.bin in ms1m-retinaface  is 0.99883 . However, the same model only got 0.99833 on the lfw.bin in faces_emore. Is there something wrong or may I ask you how many labels you have modified? Thanks.

**Comment by BingranHu** [comment]
Hi @nttstar 
I have trained a R100SE model with arcface loss on the ms1m-retinaface dataset very recently.
And the highest accuracy on the lfw.bin in ms1m-retinaface is 0.99883 . However, the same model only got 0.99833 on the lfw.bin in faces_emore. Is there something wrong or may I ask you how many labels you have modified? Thanks.

**Comment by nttstar** [comment]
@LCorleone They're both aligned by RetinaFace, so no need to do anything on training or testing datasets.

**Comment by nttstar** [comment]
@BingranHu We can get higher accuracy on LFW, CFP_FP and AgeDB_30 by the help of ms1m-retinaface.

**Comment by xizi** [comment]
@nttstar Hi, where is the download links of test data?

**Comment by nttstar** [comment]
@xizi Not ready yet.

**Comment by GewelsJI** [comment]
Does the organizing association have some limitation on the number of participants?
@nttstar

**Comment by nttstar** [comment]
@GewelsJI No

**Comment by grib0ed0v** [comment]
@nttstar When do you expect test server will be ready?

**Comment by muyouhang** [comment]
This is error reported, when I use  mx.io.ImageRecordIter to load  train.rec  , details below:
`OpenCV Error: Assertion failed (!buf.empty() && buf.isContinuous()) in imdecode_, file /home/travis/build/dmlc/mxnet-distro/deps/opencv-3.3.0/modules/imgcodecs/src/loadsave.cpp, line 637 `

Is there some problems in the provide file 'train.rec' ?

**Comment by nttstar** [comment]
@grib0ed0v Soon~

**Comment by sybyun** [comment]
I have a question about the size of parameters.
Does the 20M parameters mean total (include FC layers) size or only backbone network's size?

**Comment by nttstar** [comment]
@sybyun 
From the input layer to embedding layer. 
By the way, it is 20MB size which means 5M parameters.

**Comment by chenshen03** [comment]
The test server could not log in.

**Comment by nttstar** [comment]
@chenshen03 what's the problem? I can login successfully.

**Comment by chenshen03** [comment]
@nttstar 
The account can login normally on http://trillionpairs.deepglint.com, but there was an error when it logged on to the test server http://39.104.128.76.

![image](https://user-images.githubusercontent.com/20432947/57135854-6a4ad580-6ddd-11e9-894b-199a7a6b432d.png)

**Comment by nttstar** [comment]
@chenshen03 sign up a new account, database not shared with deepglint server.

**Comment by chenshen03** [comment]
@nttstar The database is not shared, but the system prompts "user name or email already exists" when registering. Do we need to use a different email address to register?

**Comment by nttstar** [comment]
@chenshen03 Hi, what is your old email address?

**Comment by chenshen03** [comment]
@nttstar email: chenshen@stu.xmu.edu.cn  username: Shen Chen. Thank you.

**Comment by nttstar** [comment]
@chenshen03 Can you try again to sign up a new account with chenshen@stu.xmu.edu.cn ? We have done a cleaning.

**Comment by chenshen03** [comment]
@nttstar Successfully registered, thank you!

**Comment by yeluyue** [comment]
@nttstar I couldn't find where to download  test data's pair list. Can you tell me the download link of the pair list for making the binary feature matrix (ImageNum x FeatureDim in float32) to the test server?

**Comment by nttstar** [comment]
@yeluyue Pair list is not required.

**Comment by yeluyue** [comment]
@nttstar Then how to generate the binary feature matrix? Now I have downloaded the test data and the feature_tools from the trillion Pairs, and I think the images needing to make feature matrix exists the insightface.bin. Can you tell me whether my understanding is correct?

**Comment by nttstar** [comment]
@yeluyue Please check the tutorial [here](https://github.com/deepinsight/insightface/tree/master/iccv19-challenge). I have provided the feature generation tools.

**Comment by yeluyue** [comment]
@nttstar Thank you ~

**Comment by tianxingyzxq** [comment]
Entity validate failed: user name is empty

**Comment by nttstar** [comment]
@tianxingyzxq Try logout and login again?

**Comment by NothingL** [comment]
Is data augmentation(random resize and crop) permitted?

**Comment by nttstar** [comment]
@NothingL It is allowed. But you must always use pre-aligned test dataset so normally you will get a worse result by using training time random resize and crop.

**Comment by ckybit** [comment]
hi @nttstar ，it is too slow to upload the bin file, 7 hours passed and it only upload 75%. is the right ?
how long do you take to upload the 1.8G bin file ? thank you very much.

**Comment by nttstar** [comment]
@ckybit less than 5 minutes. where are you?

**Comment by ckybit** [comment]
> @ckybit less than 5 minutes. where are you?
i'm in china and  i use company's network. i will change the network to try again,thks for your quick reply.

**Comment by zhouwei5113** [comment]
@nttstar  Is there any approach to upload test bin file on my training server directly, other than by browser?

**Comment by nttstar** [comment]
@zhouwei5113 Sorry no command line tool

**Comment by NothingL** [comment]
> @NothingL It is allowed. But you must always use pre-aligned test dataset so normally you will get a worse result by using training time random resize and crop.

thanks for clarification, is that mean we cannot resize and crop test image, even the cropped size is also 112*112 ?(means using a smaller region of face)

**Comment by yeluyue** [comment]
Is there any differences  between the feature matrices that need to be submitted for  deepglint-light and deepglint-large, or should the feature matrices be the same for both of them? @nttstar

**Comment by YongdongTan** [comment]
How do I open the .z01 and .z02 files to get the image? I am using a ubuntu system.

**Comment by nttstar** [comment]
@NothingL Right

**Comment by nttstar** [comment]
@yeluyue The shape of submission feature should always be (N, feature_dim) N=num of images. feature_dim can be 512, 256, etc..

**Comment by nttstar** [comment]
@YongdongTan Please check the testing section of [tutorial](https://github.com/deepinsight/insightface/tree/master/iccv19-challenge).

**Comment by olojuwin** [comment]
how to get Network y2 and Network r100fc?

**Comment by YongdongTan** [comment]
I executed to unzip: the command always reports zip I/O error: No such file or directory
Zip error: File not found or no read permission (iQIYI-VID-FACE.zip)

**Comment by nttstar** [comment]
@olojuwin check sample_config.py

**Comment by olojuwin** [comment]
> @olojuwin check sample_config.py

Thanks a lot.How to get  cfp_fp.bin or agedb_30.bin. I remember there used to be connected, but now I can't find it.

**Comment by nttstar** [comment]
in training dataset

**Comment by olojuwin** [comment]
It's very embarrassing，^_^

**Comment by luzai** [comment]
Hi~ I submitted two feature files, as shown in the figure below, the ROC curve demonstrate thath `r50 256 rf` model should achieve TPR@FAR1e-4=61%, but the result shown in table is the same as last model. May I ask the is there any rule for naming `filename` and `description` or anything I missed? 

![image](https://user-images.githubusercontent.com/14788650/57506387-4253f880-732e-11e9-8318-6fe4731d8d88.png)

**Comment by nttstar** [comment]
@luzai Hi, we will look into this issue soon, and also we will update a new iqiyi ground-truth.

**Comment by luzai** [comment]
> @luzai Hi, we will look into this issue soon, and also we will update a new iqiyi ground-truth.

Thank you very much for your quick response and kind remind for updating ground-truth! 

I am quite sorry because I just find there is a bug in my code. Thus, it is reasonable for the performance keeping the same as the last model, since new method is now applied in fact. After fixing it and resubmitting, there is some improvement on performance now. 

In a word, I guess the number in the table is correct and the ROC curve may have some problems.

**Comment by mlourencoeb** [comment]
Hello @nttstar. On the workshops webpage you mention the release of baseline models and corresponding training logs. Any target date for this?

Cheers

**Comment by hengck23** [comment]
for the submit test image feature, the file size is about 3.8GB. How long does it take to upload to the test server from overseas?

It seems to be very slow?

**Comment by MinZhaoLin** [comment]
@nttstar Hellow, how can I get the FLOPs of my models?

**Comment by ZHAIXINGZHAIYUE** [comment]
where to get network y2?

**Comment by nttstar** [comment]
@MinZhaoLin It will output FLOPs when you start training. See ``common/flops_counter.py`` for detail.

**Comment by nttstar** [comment]
@ZHAIXINGZHAIYUE See ``recognition/sample_config.py``

**Comment by nttstar** [comment]
@hengck23 How long did you take? Where're you?

**Comment by ZHAIXINGZHAIYUE** [comment]
@nttstar @MinZhaoLin 

> @MinZhaoLin It will output FLOPs when you start training. See `common/flops_counter.py` for detail.

just count conv and fc?

**Comment by nttstar** [comment]
@ZHAIXINGZHAIYUE any else operator you want to use?

**Comment by hengck23** [comment]
@nttstar

>> @hengck23 How long did you take? Where're you?
I am in singapore. I cannot upload for test image (feature file size is 3.8 Gb) because of connection timeout. for test video, the feature file size is about 500 mb and it takes about 1 hour for me.

**Comment by nttstar** [comment]
@hengck23 My friend from Singapore said he had no problem with submitting. I guess your network provider has some connection issue with AliCloud.

**Comment by hengck23** [comment]
i am starting a discussion slack team at : https://join.slack.com/t/private-5ay1415/shared_invite/enQtNjI4NDA2MjUzNzkzLThlZDIxODBlMjZiMmIzZGVhMDRiNjExZGY2YzE5ZTEyOTczZDUxZjc0OWUyNTQzNDZkMDYxNjNkYWFkODY5OTc

Please feel free to join. Objectives of the slack team:

1. share some baseline code and results.

2. discussion possible strategy and papers, etc to achieve best results. 

As a start, i am open sourcing by pytorch baseline code and trained model below. Join the slack for more information!

![ir50](https://user-images.githubusercontent.com/5112846/57692893-7e62c280-767a-11e9-9f6e-448f0bfc581d.png)

**Comment by LeonLIU08** [comment]
Why I cannot find the iQIYI-VID-FACE.z01, iQIYI-VID-FACE.z02, and iQIYI-VID-FACE.zip files from the iQIYI AI completion platform? It seems that they have changed the name of the data. 
So how to get the video test data?

**Comment by nttstar** [comment]
@LeonLIU08 I can find it at http://challenge.ai.iqiyi.com/data-cluster

**Comment by LeonLIU08** [comment]
> @LeonLIU08 I can find it at http://challenge.ai.iqiyi.com/data-cluster

Thanks for your quick response. The website jumping is quite confusing after the registration.

**Comment by grib0ed0v** [comment]
@nttstar I've got upload error, when tried to upload 1.8 GB feature file from Russia. Any suggestion?

**Comment by mariolew** [comment]
@nttstar The submission contains features only, so how you verify the model complexity of the submission?

**Comment by hengck23** [comment]
> @nttstar The submission contains features only, so how you verify the model complexity of the submission?

i suggest all team to submit a  onnx model to the submission server, then the organizer can make an automatic script to compute the flops and ouput on leaderboard. It can be based on the original count_flops.py of mxnet

**Comment by nttstar** [comment]
@mariolew @hengck23 We will make sure the leader submissions should be reproducible.

**Comment by nttstar** [comment]
@grib0ed0v Try logout and login again

**Comment by hengck23** [comment]
https://ibug.doc.ic.ac.uk/resources/lightweight-face-recognition-challenge-workshop/

"The upper bound of model size is 20MB."

i suppose this is number of parameters and not storage in megabytes? (1 float = 4 bytes)

**Comment by nttstar** [comment]
@hengck23 20MB is the fp32 storage size, and 5M for number of parameters.

**Comment by mariolew** [comment]
@nttstar The submission of feature size is limited to 5GB, which I think is not enough for video features.

**Comment by nttstar** [comment]
@mariolew video feature file should be smaller

**Comment by hengck23** [comment]
@mariolew

you have to combine image_features to video_feature, e.g.

video_feature = average(image_features)

or 

video_feature = median(image_features)


or 

video_feature = max(image_features), etc ...

**Comment by lambdaflow** [comment]
Is there any identity label in the training set? @nttstar

**Comment by ShijieVVu** [comment]
@nttstar Would you be sharing the y2 and r100fc baseline pre-trained models to the public? Thanks.

**Comment by hengck23** [comment]
> @hengck23 My friend from Singapore said he had no problem with submitting. I guess your network provider has some connection issue with AliCloud.

i now set up AWS EC2 cloud computer to do the upload to the submission server, I am getting uploading speed of about 50 to 100 kbits/sec for ec2 in singapore and hongkong. (the network limit of my ec2 is 5 Mbit/sec)

Can anyone report there upload speed and their location? I would like to setup cloud PC there.

**Comment by PistonY** [comment]
@hengck23 How do you generate submit bin file?Do you use this repo `gen_image_feature.py` ?
I don't know how to generate this easier cause I use python3 and Gluon....
@nttstar Or could you please provide a common format for the bin file?

**Comment by hengck23** [comment]
> @hengck23 How do you generate submit bin file?Do you use this repo `gen_image_feature.py` ?
> I don't know how to generate this easier cause I use python3 and Gluon....
> @nttstar Or could you please provide a common format for the bin file?

yes, i use code from gen_image_feature.py. you can just use the copied code:

```
import struct
def write_binary(binary_file, data):
    rows, cols = data.shape
    with open(binary_file, 'wb') as f:
        f.write(struct.pack('4i', rows, cols, cols*4, 5))
        f.write(data.data)


def read_binary(binary_file):
    with open(binary_file, 'rb') as f:
        bytes = f.read()

    rows, cols, d0, d1 = struct.unpack_from('4i',bytes)
    assert(d0==cols*4)
    assert(d1==5)

    offset = len(struct.pack('4i', rows, cols, cols*4, 5))
    data = np.frombuffer(bytes, dtype=np.float32, offset=offset)
    data = data.reshape(rows, cols)
    return data
```

data is an np array with array.shape = (num_of_face_images, feature_dim).

and fyi, mxnet insightface is python3 compatible after small changes (e.g. xrange to range, and change python3 pickle load function to read python2 created pickle files)

**Comment by PistonY** [comment]
@hengck23 Thanks anyway.

**Comment by nttstar** [comment]
@hengck23 Thank U.

**Comment by believeitmyway** [comment]
Is L2 normalization of embedding vectors required in this challenge?
L2 normalization is performed in the ArcFace, but there is the other methods that L2 normalization of embedding vectors is not performed.

**Comment by PistonY** [comment]
@believeitmyway you need not to submit L2 normalization features,I think the task should use `euclidean distance` but not `cosine distance` to get similarity, so the L2 normalization is not necessary.

**Comment by PistonY** [comment]
I share a Gluon/Pytorch(need a little change) way to get the submission bin file for image/video(later update) task.
https://gist.github.com/59d312ff89ddcb891a812fc001e15d62
@hengck23 Could you please test it if could get totally same file with `gen_image_feature.py`, it should be very easy refactor to Pytorch style. if you If you are convenient.

**Comment by nttstar** [comment]
@believeitmyway @PistonY Evaluation metric is cosine.

**Comment by PistonY** [comment]
So it’s only support margin based loss...

**Comment by nttstar** [comment]
No, all embedding based approaches should be fine. (Softmax, Triplet Loss, Arcface, etc..)

**Comment by chichan01** [comment]
Hi, 
  Is there any better solution to upload the result to testing server?  I am not able to upload it in UK, it always has time-out problem...  The best record is only 10% of 3.8G data.

**Comment by believeitmyway** [comment]
@nttstar @PistonY Thanks.
So distance based loss like contrastive loss and triplet loss is not available in this challenge, right?

**Comment by believeitmyway** [comment]
If evaluation metric is simple L2 distance, both of distance based loss and angular based loss is available.

**Comment by nttstar** [comment]
@believeitmyway We should always compare l2 normed embeddings, for all algorithms you have metioned, contrastive loss, triplet loss, arcface. So no problem of cosine evaluation metric.

**Comment by believeitmyway** [comment]
@nttstar Recent methods perform L2 normalization as you say.
But I do not think that it is essential for the future.
When targeting a metric space different from the conventional method, L2 normalization may not be essential.
But if evaluation metric has already been decided, I will follow it.
Thank you.

**Comment by nttstar** [comment]
@believeitmyway Yes L2 distance metric is more general. We will use cosine this time, as we already have many submissions, Thank you for your advice!

**Comment by PistonY** [comment]
@believeitmyway you just need to set embedding layer's `bias=False` and pass L2 normalization, it should be ok with `cosine distance`

**Comment by liujiaheng** [comment]
Has anyone validated the flops of r100? I just got 12.1GFLOPS.

**Comment by nttstar** [comment]
@liujiaheng https://github.com/deepinsight/insightface/blob/master/common/flops_counter.py

**Comment by mlourencoeb** [comment]
@nttstar would it be possible to upload baselines in Dropbox? Thanks.

**Comment by jingyi1997** [comment]
Hi! My training speed is about 530 samples/sec. However, the speed recorded in the log is about 700 samples/sec. How can I speed up my training?

**Comment by yu4u** [comment]
Thank you for interesting competition!

> (1) All participants must use this dataset for training without any modification (e.g. re-alignment or changing image size are both prohibited).

I'd like to confirm the above rule.
It means that ALL data augmentation including random crop or contrast change in training are prohibited, right?
How about normalization?

**Comment by nttstar** [comment]
@yu4u The alignment method should be kept, no change. But color jittering is allowed. What do you mean by normalization?

**Comment by yu4u** [comment]
@nttstar Thank you for immediate reply!

> But color jittering is allowed

Oh, I see.
Sorry, I meant standardization in terms of mean and variance by normalization.
This of course should be allowed as color jittering is allowed (and seems to be done in reference code)

**Comment by PistonY** [comment]
Changing input image size is prohibited?

**Comment by nttstar** [comment]
@PistonY Training and testing images should not be changed like re-alignment or something else. Training time data augmentation is allowed but only flipping test is allowed in testing time.

**Comment by hengck23** [comment]
@nttstar 

Assuming we do distillation or domain shift. Are the following allowed?

1) train a teacher network using train. Treat test data as unlabeled, and use teacher to distill 'train+test' to student network.

2) Treat test data as unlabeled target domain and train as labeled source domain. Learn features that minimize "feature difference" of source and target domain

**Comment by mariolew** [comment]
@hengck23 (1) All participants must use this dataset for training without any modification (e.g. re-alignment or changing image size are both prohibited). 
(2) Any external dataset is not allowed.
I don't think it's allowed.

**Comment by nttstar** [comment]
@hengck23 Test data is not allowed to be used in any training process.

**Comment by mariolew** [comment]
@nttstar In this repo, FLOPS are counted as a base of 1000, not 1024, so, there might be some models of 1.03G(base 1000) FLOPS, but it is not really 1.03G(base 1024).

**Comment by cbstyle** [comment]
@nttstar hi, how can i get a validation dataset? Splitting from the training datasets or use lfw datasets?

**Comment by nttstar** [comment]
@mariolew Not a big problem, we will validate all top submissions after ddl.

**Comment by nttstar** [comment]
@cbstyle LFW, CFP, AgeDB in training dataset.

**Comment by MrGF** [comment]
@nttstar  Why it has to multiply by 2 in the calculation of FLOPs? 
![image](https://user-images.githubusercontent.com/6449172/58326049-46b7ff80-7e5e-11e9-9a1e-7ac33aaf44b3.png)

**Comment by nttstar** [comment]
@MrGF Same as opencv dnn

**Comment by LeonLIU08** [comment]
Hi, I have successfully submitted one result for "deepglint-light" to the server, but I still cannot get any result shown from the webpage. How long it will take to provide the performance?

**Comment by SoftwareGift** [comment]
![image](https://user-images.githubusercontent.com/23412962/58393807-eea61680-8072-11e9-99ce-443a6c16eecc.png)
We have created a discussion group for the competition and welcome everyone to join us.

**Comment by jingyi1997** [comment]
Hi! I use my model to extract features and submit it on the test server. The acc is 0. However, my model's performance on LFW is normal. I have also tried to submit the features to the official trillionpair's website and the result is also correct. I can not figure out the possible reason for that. My file is in the binary format and the data is  1,862,120 * 256 in float32.

**Comment by nttstar** [comment]
@jingyi1997 I'm not quite sure about your question. But many others can get correct score by 256 or 512 embeddings.

**Comment by cham-3** [comment]
@nttstar Can I train models with the IQIYI_DIV_TRAIN for the video test?

**Comment by nttstar** [comment]
@cham-3 no

**Comment by luzai** [comment]
The rules says `Modification (e.g. re-alignment or resize) on test images is not allowed`. May I ask is it permitted to use deconvolution or upsampling to increase the resolution of the features or even the input images? Upsampling seems to be an essential part of high resolution net. Or is there some restrictions on how to use this operations?

**Comment by nttstar** [comment]
@luzai Do not resize on images. You can use upsampling or deconvolution but make sure to add the FLOPs of deconvolution operator which is not contained in current FLOPs counter.

**Comment by grib0ed0v** [comment]
For PyTorch users, you can use [flops-counter](https://github.com/sovrasov/flops-counter.pytorch).

However note that it outputs **MACs not FLOPs** - so you need to have **model less than 500 MMac**

**Comment by nttstar** [comment]
Hi all, we will update the deepglint(image testset) evaluation metric from FPR=e-8 to FPR=e-6, to get a more stable accuracy. This is the final decision.

**Comment by xizi** [comment]
@nttstar where is video_filelist.txt?

**Comment by nttstar** [comment]
@xizi in testdata-image.zip

**Comment by xizi** [comment]
@nttstar ok.

**Comment by nttstar** [comment]
We have created a wechat group for Chinese discussion:
Click for QR code: https://github.com/deepinsight/insightface/blob/master/resources/lfr19_wechat1.jpg 

Any idea for where we can build a chat group for English?

**Comment by grib0ed0v** [comment]
@nttstar Maybe whatsapp?

**Comment by nttstar** [comment]
Slack group for English:
https://join.slack.com/t/insightface/shared_invite/enQtNjU0NDk2MjYyMTMzLTIzNDEwNmIxMjU5OGYzYzFhMjlkNjlhMTBkNWFiNjU4MTVhNTgzYjQ5ZTZiMGM3MzUyNzQ3OTBhZTg3MzM5M2I

**Comment by nttstar** [comment]
@grib0ed0v Thanks, but it's a little bit complex for me to download whatsapp app..

**Comment by sarathknv** [comment]
My uploads are terminating after some time saying ```Upload error. Please refresh and try again```. Any idea why this is happening?

**Comment by nttstar** [comment]
@sarathknv Hi, sorry for the inconvenience. Have you tried again? Where are you(country and city)?

**Comment by nttstar** [comment]
@sarathknv You can try to logout and re-login before submitting the results.

**Comment by alphanerdj** [comment]
Is there a recommended way to measure FLOPs in tensorflow? I saw that your sample code is for mxnet. Is that equivalent of 
    opts = tf.profiler.ProfileOptionBuilder.float_operation()    
    flops = tf.profiler.profile(graph, run_meta=run_meta, cmd='op', options=opts)

**Comment by gehaocool** [comment]
@nttstar Hi, I have a question.
How do you make sure the participants strictly follow the rules such as no extra data and no modification on the data?
You've mentioned above that you'll make sure the leading results are reproducible, so how to do this?
Should the participants upload model and training code?

**Comment by nttstar** [comment]
@gehaocool We will make sure about this and let you all know about it. No worries.

**Comment by cham-3** [comment]
@nttstar Hi, The gpu is out of memory when I use gen_image_feature.py to extract features with a batch_size 64 same as training phase. is it normal?

**Comment by vitoralbiero** [comment]
@nttstar Hi, is it allowed to use less training data than the full dataset? e.g., use only 50% of the training data. Thank you!

**Comment by caocuong0306** [comment]
Hi @nttstar, I have several questions.

1. How can we obtain raw 112x112 face images extracted by RetinaFace? I'm using Tensorflow to perform training and validation, so I need to access to raw images.

2. As @alphanerdj asked, is there standard FLOPs calculation function for Tensorflow (which is similar to current MXNet-based function)

Thank you.

**Comment by nttstar** [comment]
@vitoralbiero It is allowed. But you can not use external datasets/models to determine the subset selection.

**Comment by nttstar** [comment]
@caocuong0306

**Comment by caocuong0306** [comment]
@nttstar ,

1. I did unpack training images from .rec files, but validation sets (LFW, agedb, cfp) were packed in .bin files.

Can you provided direct access to raw images for validation purpose? 

Thank you.

**Comment by nttstar** [comment]
@caocuong0306 You may refer to the 1:1 verification code to see how it be loaded. Then you will know how to unpack it.

**Comment by paulcx** [comment]
@nttstar Is flops-counter approved by official for calculating FLOPs value which is just half of Mac value?

**Comment by nttstar** [comment]
@paulcx Please refer to our FLOPs counter.

**Comment by nttstar** [comment]
We have updated the ground-truth of glint dataset and all incoming submission will get the updated accuracy. We will also make update on top existing submissions soon.

**Comment by sybyun** [comment]
@nttstar I found very strange result from the new light image ranking list.
yajundang_test1's model name is exactly same as my image feature name of 6/13 ('20190613102605_l5_dw_conv6_fc512_image47').
I don't know how it is possible. 
I didn't follow public naming rule so it can be same by accident and the first part of name couldn't be manually given.
Can you explain it?

**Comment by nttstar** [comment]
@sybyun  we will check it asap, all feature files are kept, dont worry

**Comment by nttstar** [comment]
@sybyun yujundang is our test account to submit existing features, it should be no problem.

**Comment by alphanerdj** [comment]
I've been trying to upload feature file multiple times since yesterday but the test server is always stuck at 1% upload. I'm in the US. Can you help check that upload is working?

**Comment by nttstar** [comment]
@alphanerdj Can you try WIN/MAC + Chrome to submit the feature? And try using another computer if it doesnot work.

**Comment by alphanerdj** [comment]
It started working, thanks!

**Comment by nttstar** [comment]
LB scores are delaying as too many submissions, please wait.

**Comment by believeitmyway** [comment]
Can we use your baseline model as a pre-trained model?
I'd like to try finetuning of the model with different losses or parameters.

**Comment by nttstar** [comment]
@believeitmyway Yes you can. But the pretrained models from our ArcFace paper are not allowed.

**Comment by believeitmyway** [comment]
Thanks!

**Comment by nemonameless** [comment]
@nttstar  please update the Chinese wechat QR code  , thanks

**Comment by nttstar** [comment]
We will accept the solutions whose FLOPs<=1024^3

**Comment by alphanerdj** [comment]
What about model size?

**Comment by nttstar** [comment]
@alphanerdj comment updated.

**Comment by nttstar** [comment]
funding pool updated: now $17000

**Comment by sefaburakokcu** [comment]
> Hi! I use my model to extract features and submit it on the test server. The acc is 0. However, my model's performance on LFW is normal. I have also tried to submit the features to the official trillionpair's website and the result is also correct. I can not figure out the possible reason for that. My file is in the binary format and the data is 1,862,120 * 256 in float32.
> ![Screenshot from 2019-05-27 16-58-02](https://user-images.githubusercontent.com/41174283/58408561-a30a6180-80a0-11e9-86bb-caa0a4a5251f.png)
> ![Screenshot from 2019-05-27 16-57-50](https://user-images.githubusercontent.com/41174283/58408563-a3a2f800-80a0-11e9-8ba2-8334ab746eaa.png)

@jingyi1997  I have encountered the same problem. Were you able to understand the problem and solve it?

**Comment by alphanerdj** [comment]
The upload server is very very slow and works only once in a while. Even video model can take hours to load and then still fail towards end of upload. Tried different of browsers/OS and computers. Is there any alternate way/server to submit models?

**Comment by alphanerdj** [comment]
Should imageset and video set features be generated using same model to be considered in final ranking?

**Comment by nttstar** [comment]
@alphanerdj Using different models is allowed.

**Comment by boomberung** [comment]
Hello! I tried to download the solution many times from work and from home wiht Windows and Linux, but the download does not go beyond 1%, can you fix it before the deadline?

**Comment by garaevnick** [comment]
i have a same problem, i cant submit from my PC whit Ubuntu 19.04 + Chrome, please help me submit or move deadline please.

**Comment by ms-lilibeth** [comment]
@nttstar same problem as @boomberung has
Ubuntu 18.04 + Chrome
Could you, please, help me? Or can deadline be postponed?

**Comment by nttstar** [comment]
@boomberung You mean submit?

@boomberung @ms-lilibeth You can send your feature files to insightface.challenge@gmail.com, before deadline, with your username and password. One for each track at most.

**Comment by luzai** [comment]
@nttstar, Can we use any cloud platform for upload feature files and attach the link to the mail? Or can we send md5 of feature file and upload after deadline?

**Comment by nttstar** [comment]
@luzai Yes you can use google drive or dropbox, before DDL only. I think there is no link problem for Chinese participants.

**Comment by nttstar** [comment]
Game closed. We will send emails to top participants.

**Comment by nttstar** [comment]
We have sent final check emails to top 20 participants. If you do not receive it, please contact us.

**Comment by jinwu07** [comment]
@nttstar Can the test server still be used for the purpose of valuation after competition is over?
Thanks,

**Comment by Wxsang** [comment]
@nttstar I need to do a practical work in camera.
ms1m-retinaface is aligned by retina, the align algorithm is same as '/src/common/face_preprocess.py' ?
Thank for your help.

**Comment by nttstar** [comment]
Leaderboard released. See [http://insightface.ai/build/lfr19/leaderboard.html](http://insightface.ai/build/lfr19/leaderboard.html)

**Comment by borisgribkov** [comment]
@nttstar Are you going to provide online translation of the сhallenge on Monday? Thanks!

**Comment by nttstar** [comment]
@borisgribkov hmm...no

**Comment by nttstar** [comment]
I can post some pictures here if you need.

**Comment by borisgribkov** [comment]
bad news, would be nice to watch and hear about winner solutions.... that is why I asked... 
about a month ago I asked ICCV organizers the same question about online YouTube translation and they replied: "yes" ( and we decided not to fly to Seoul :) ), seems like it was about conference talks not about workshops and tutorials.

**Comment by nttstar** [comment]
@borisgribkov Maybe I will open a stream at [http://live.bilibili.com/13394860](http://live.bilibili.com/13394860) on next Monday, but no translation.

**Comment by borisgribkov** [comment]
Thanks a lot! Regarding language, will it be in Chinese? ;)

**Comment by nttstar** [comment]
Ah, it should be English.

**Comment by borisgribkov** [comment]
Great! Thanks!

**Comment by borisgribkov** [comment]
Hi @nttstar ! We are idiots, because of time shift we missed it :((((  Maybe you or someone record something? Thanks a lot....

**Comment by HaoLiuHust** [comment]
how to get the champion's  solution?

**Comment by versus-x** [comment]
@borisgribkov https://www.youtube.com/playlist?list=PLiG8_90geVeVxRZ7hducf7M4ZHWGJcBEs

**Comment by borisgribkov** [comment]
@versus-x Thanks a lot!

**Comment by dexception** [comment]
@nttstar Are you going to see any release for lightweight models ?

**Comment by xmuszq** [comment]
@nttstar 
where can i see the published papers in this workshop?

**Comment by mlourencoeb** [comment]
Hi @nttstar, do you have the training logs for the baselines provided in the paper? If so, could you gently upload them in dropbox or google drive please?

**Comment by xszg007** [comment]
@nttstar HI, I want to test my modle under the framework of this challenge but it seems that the test server doesn't work now.  How can I workout the modle performance offline? Can I get the result by following the training and testing rules in the front page without the server?
